src/controller/SetUpCodePairer.cpp

```diff
@@ -62,7 +62,8 @@ CHIP_ERROR SetUpCodePairer::Connect(RendezvousInformationFlag rendezvousInformat
         VerifyOrReturnError(searchOverAll || CHIP_NO_ERROR == err, err);
     }
 
-    if (searchOverAll || rendezvousInformation == RendezvousInformationFlag::kOnNetwork)
+    // We always want to search on network because any node that has already been commissioned will use on-network regardless of the
+    // QR code flag.
```

Comment:
Can you move the following code after the `StartDiscoverOverSoftAP` case and update a little then ?
For example it can be rewritten as: 
```
// By default, always search on network because any node that has already been commissioned
// will use on-network regardless of the flag.
if (CHIP_NO_ERROR == (err = StartDiscoverOverIP(discriminator, isShort)))
{
        isRunning = true;
}
VerifyOrReturnError(searchOverAll || CHIP_NO_ERROR == err, err);
```

Reply:
I changed the order in the next commit, but I'm not totally sure I understand - the order is different, but the logic is the same because these are independent blocks so it doesn't simplify anything. I might have misunderstood what you wanted here.